haskell/private/haddock_wrapper.sh

```diff
@@ -36,4 +36,8 @@ do
     fi
 done
 
+# Override TMPDIR to prevent race conditions on certain platforms.
+mkdir haddock-tmp
+TMPDIR=haddock-tmp
```

Comment:
I do not understand the solution and you do not explain it in the description. You set `TMPDIR` to a constant name, `haddock-tmp`, how does it help to avoid name clashes?

Another question: what happens when `haddock-tmp` already exists?

And finally, is this code exception-safe? What if while Bazel is working I hit control-C and shut it down. I don't know exactly, but perhaps the script will receive a signal of some sort. So that signal could arrive after `mkdir` but before `rmdir` and so the directory won't be cleaned up.

I suggest we use `mktemp` which can randomize names by itself and `trap` for signal handling perhaps? What do you think?

Reply:
AIUI this script is always run in a unique directory per-target. It seemed to be different on each bazel build, so I assumed it to be a clean temporary directory. I don't really know bazel, so maybe that is not quite correct, but this fixed the bug without dealing with cross-platform `mktemp` invocation

`mktemp` is fine by me if you prefer it. I'll amend the PR soon.